### DISCUSSION TITLE
Print all rust code (inside project) executed by a single test

### DISCUSSION THREAD

**Discussion by Anton-4** [xd009642/tarpaulin#1507] [Q&A] [answered]
I'd like to compare the code executed by a test for two different commits. So I'm looking for a way to print every line of rust code executed by that test, excluding dependencies. Because code coverage is quite similar in nature, I thought I'd ask here for tips. Do you know of a tool that can do this or could I try to hack/use parts of tarpaulin to make this happen?

I'm obviously aware of stepping with a debugger but all the jumping back and forth is slow, and I'd prefer to be able to scroll through two "execution traces" side by side.

**Comment by xd009642** [comment, marked as answer]
So the json output of tarpaulin gives you the file paths, lines and hits. If you have that json for two different commits you could compare it directly or use it plus a script to print out things the way you like. I also generate entirely self-contained HTML reports and in the past I've generated a report per commit and just stuck two browser windows side by side if that's acceptable

**Comment by Anton-4** [comment]
Cool, thank you @xd009642 :heart: 

Can I run tarpaulin on a single test when there are multiple tests in a package? I tried some variations that work with `cargo test` but I could not get it to work.

**Reply by xd009642** [reply]
`cargo tarpaulin -- $TEST_NAME` should work, the same as cargo test
